tests/components/FilterSearch.test.tsx

```diff
@@ -0,0 +1,254 @@
+import { FilterSearch } from '../../src/components/FilterSearch';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import * as searchOperations from '../../src/utils/search-operations';
+import { mockAnswersActions, spyOnActions } from '../__utils__/mocks';
+import { labeledFilterSearchResponse, unlabeledFilterSearchResponse, noResultsFilterSearchResponse } from '../../tests/__fixtures__/data/filtersearch';
+import { Matcher } from '@yext/answers-headless-react';
+
+jest.mock('@yext/answers-headless-react');
+const actions = spyOnActions();
+
+const setFilterOption = jest.fn();
+const setOffset = jest.fn();
+const searchFieldsProp = [{
+  fieldApiName: 'name',
+  entityType: 'ce_person'
+}];
+
+const expectedFilterSearchFields = [{
+  entityType: 'ce_person',
+  fetchEntities: false,
+  fieldApiName: 'name'
+}];
+
+describe('search with section labels', () => {
+  beforeEach(() => {
+    mockAnswersActions({
+      setFilterOption,
+      setOffset,
+      executeFilterSearch: jest.fn().mockResolvedValue(labeledFilterSearchResponse)
+    });
+  });
+
+  it('renders the filter search bar and the "Filter" label', () => {
+    render(<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const label = 'Filter';
+    const labelElement = screen.getByText(label);
+    const searchBarElement = screen.getAllByRole('textbox');
+
+    expect(labelElement).toBeDefined();
+    expect(searchBarElement.length).toBe(1);
+  });
+
+  it('displays characters typed in search bar correctly',() => {
+    render(<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+
+    userEvent.type(searchBarElement, 'n');
+    expect(searchBarElement).toHaveValue('n');
```

Comment:
your idea of condensing the "n" and "a" type calls

Reply:
combined